### PR TITLE
HARP-13369: update package dependencies

### PR DESCRIPTION
**Pull request by dbacarel**
Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

**Comment by dbacarel** [comment]
@harpgl_bot retest this please

**Comment by dbacarel** [comment]
@harpgl_bot retest this please

**Comment by dbacarel** [comment]
@harpgl_bot retest this please

**Comment by nzjony** [comment]
@harpgl-bot retest this please

**Comment by nzjony** [comment]
@dbacarel , when you use @, you can find the bot, just search "@harp", that might help because "@harpgl_bot" doesn't exist.

**Comment by dbacarel** [comment]
interesting, thank you @nzjony. Maybe it was a coincidence but also via "harpgl_bot" it triggered the retest 🤷🏻‍♂️

**Comment by nzjony** [comment]
> interesting, thank you @nzjony. Maybe it was a coincidence but also via "harpgl_bot" it triggered the retest 🤷🏻‍♂️

Hmm.. that reminds me, I remember that the @harpgl-bot is actually redundant, and that just "retest this please" is needed, see: https://plugins.jenkins.io/ghprb/ I forgot :(

**Comment by dbacarel** [comment]
retest this please

File: .eslintrc.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
         "project": "./tsconfig.json"
     },
     "rules": {
-        "prettier/prettier": "error",
+        "prettier/prettier": ["error", {
```
Comment (FraukeF) [inline .eslintrc.json:8]: why did you add this condition, did the default of eslint change?

File: .prettierrc.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
     "printWidth": 100,
-    "tabWidth": 4
+    "tabWidth": 4,
+    "trailingComma": "none"
```
Comment (FraukeF) [inline .prettierrc.json:4]: same here, did the default change?

File: @here/harp-atlas-tools/src/AtlasGenerator.ts [hunk 3]
```diff
@@ -320,7 +320,7 @@ async function readAndProcessImage(
     logger.log(LogLevel.DEBUG, `Processing file: '${inFile}`);
     const imageDecoder = await imageFactory.loadImage(inFile);
     let imageOutput: ImageEncoder = imageDecoder.copy();
-    operations.forEach(imageProcess => {
+    operations.forEach((imageProcess) => {
```
Comment (FraukeF) [inline @here/harp-atlas-tools/src/AtlasGenerator.ts:323]: why is this needed now, it changes almost all files, is there a way to configure the new packages to work as before?

**Review by FraukeF — COMMENTED** [review]
Please double check to configure the packages, or maybe its only your local copy to not do all these reformattings

File: .eslintrc.json (hunk 1 above)
Comment (dbacarel) [reply .eslintrc.json:8]: I changed the rule because Prettier v2 default for trailing commas changed and the eslint fails without this extra rule.
Although I find it strange since I explicitly set "trailingComma": "none" in .prettierrc.json

File: .prettierrc.json (hunk 2 above)
Comment (dbacarel) [reply .prettierrc.json:4]: Yes, Prettier v2 has now trailing commas set to "es5" by default, before it was "none"
https://prettier.io/docs/en/options.html#trailing-commas

File: @here/harp-atlas-tools/src/AtlasGenerator.ts (hunk 3 above)
Comment (dbacarel) [reply @here/harp-atlas-tools/src/AtlasGenerator.ts:323]: Prettier v2 has now arrow-function-parentheses set to "always" by default, before it was "avoid".
https://prettier.io/docs/en/options.html#arrow-function-parentheses

I can add an additional rule in .prettierrc.json and remove those parentheses in case we want to stick with the previous style

Comment (FraukeF) [reply @here/harp-atlas-tools/src/AtlasGenerator.ts:323]: yes, please, lets stick to the previous formatting rules as far as possible.

**Review by gairik — COMMENTED** [review]
`
"prettier": "prettier -l "**/*.ts" "**/*.tsx" "**/*.json"",
"prettier:fix": "prettier --write "**/*.ts" "**/*.tsx" "**/*.json"",
`

File: @here/harp-atlas-tools/src/AtlasGenerator.ts (hunk 3 above)
Comment (dbacarel) [reply @here/harp-atlas-tools/src/AtlasGenerator.ts]: I added the rule to disable the parenthesis.
However, I couldn't do much about the space after the keyword 'function' on function expressions, e.g. `const foo = function () {..}`.
They don't seem to provide a way to suppress the rule - https://prettier.io/blog/2020/03/21/2.0.0.html#other-changes

**Comment by dbacarel** [comment]
retest this please

**Review by FraukeF — COMMENTED** [review]
This change is still enormous due to the prettier stuff, could you split it into two seperate PRs one without the prettier v2 update and one for the prettier v2 update

**Comment by dbacarel** [comment]
Good point, alright, I excluded Prettier's update from this PR. I'll create a separate one for it

File: .eslintrc.json [hunk 4]
```diff
@@ -41,7 +41,7 @@
         "@typescript-eslint/no-non-null-assertion": "off",
         "@typescript-eslint/explicit-function-return-type": "off",
 
-        "simple-import-sort/sort": "error",
+        "simple-import-sort/imports": "error",
```
Comment (FraukeF) [inline .eslintrc.json:44]: why did this change?

File: @here/create-harp.gl-app/package.json [hunk 5]
```diff
@@ -43,7 +43,7 @@
         "log-symbols": "^4.0.0"
     },
     "devDependencies": {
-        "mocha": "^7.2.0"
+        "mocha": "8.2.1"
```
Comment (FraukeF) [inline @here/create-harp.gl-app/package.json:46]: please use ^8.2.1 (higher then version for all the updated packages)

Comment (FraukeF) [reply @here/create-harp.gl-app/package.json:46]: also for all the following

File: @here/harp-examples/decoder/custom_decoder.ts [hunk 6]
```diff
@@ -30,9 +30,8 @@ import { CUSTOM_DECODER_SERVICE_TYPE } from "./custom_decoder_defs";
 const MIN_GEOMETRY_HEIGHT = -100;
 
 export // snippet:custom_datasource_example_custom_decoder.ts
-class CustomDecoder extends ThemedTileDecoder
-// end:custom_datasource_example_custom_decoder.ts
-{
+class CustomDecoder extends ThemedTileDecoder {
```
Comment (FraukeF) [inline @here/harp-examples/decoder/custom_decoder.ts:33]: will this throw an error if not changed?

File: @here/harp-examples/src/datasource_custom.ts [hunk 7]
```diff
@@ -108,9 +108,8 @@ export namespace CustomDatasourceExample {
     const MIN_GEOMETRY_HEIGHT = -100;
 
     // snippet:custom_datasource_example_custom_data_provider.ts
-    class CustomDataProvider extends DataProvider
-    // end:custom_datasource_example_custom_data_provider.ts
-    {
+    class CustomDataProvider extends DataProvider {
```
Comment (FraukeF) [inline @here/harp-examples/src/datasource_custom.ts:111]: same here, is  this a necessary reformatting?

File: @here/harp-map-controls/lib/MapAnimations.ts [hunk 8]
```diff
@@ -452,7 +452,9 @@ export class CameraPanAnimation extends CameraAnimation {
             })
             .onUpdate(({ latitude, longitude, altitude }) => {
                 this.mapView.geoCenter = new GeoCoordinates(latitude, longitude, altitude);
-                this.mapView.camera.position.z = altitude;
+                if (altitude) {
```
Comment (FraukeF) [inline @here/harp-map-controls/lib/MapAnimations.ts:455]: this will not set the altitude if its 0, this changes the former behaviour... why is this change needed?

File: @here/harp-mapview/test/UtilsTest.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable simple-import-sort/sort */
+/* eslint-disable simple-import-sort/imports */
```
Comment (FraukeF) [inline @here/harp-mapview/test/UtilsTest.ts:1]: why do we disable the sort here at all, maybe we can remove theline?

File: @here/harp-materials/lib/MapMeshMaterials.ts [hunk 10]
```diff
@@ -251,7 +251,7 @@ export function hasExtrusionFeature(material: any): material is ExtrusionFeature
     return "extrusionRatio" in material;
 }
 
-namespace DisplacementFeature {
+namespace DisplacementFeatureNS {
```
Comment (FraukeF) [inline @here/harp-materials/lib/MapMeshMaterials.ts:254]: why change the name?

**Review by FraukeF — COMMENTED** [review]
Thanks for removing the prettier update, its already a bit better, but I still think this change is way to big, if it causes issues later on, it will be difficult to break it down to where it actually came from... e.g. the yarn.lock file changes are unreviewable :( 

Please update always to ^<version> not to fixed ones ...

I am not 100% sure how a good split would look, one package one PR, seems to create a lot of PRs... 

I could imagine something like: 
maybe one PR with the testing devDependencies like mocha, sinoon, chai....
another PR with the packaging utilities the webpack, webpack-merge, ts-loader ... stuff

one for the tween update
one for the olp-sdk update
one for the eslint related stuff

let me know what you think

**Comment by dbacarel** [comment]
>> Please update always to ^ not to fixed ones ...

Sure, I used `./scripts/reset-dependency-version` to update the packages and it set to a fixed version (so, without ^). I thought it was the intended approach. Since it's not, I'll add the ^ manually.

>> [...] it will be difficult to break it down to where it actually came from... e.g. the yarn.lock file changes are unreviewable :(

Agree.

I'll create individual PRs for the following:

- eslint related packages
- packing utils
- tween
- olp-sdk

After that, I'll revert the changes related to these updates from this PR and see if it becomes of a reasonable size.

**Comment by FraukeF** [comment]
> > > Please update always to ^ not to fixed ones ...
> 
> Sure, I used `./scripts/reset-dependency-version` to update the packages and it set to a fixed version (so, without ^). I thought it was the intended approach. Since it's not, I'll add the ^ manually.

you can use the script also with the ^version, so you dont have to do this manually

**Comment by dbacarel** [comment]
>> you can use the script also with the ^version, so you don't have to do this manually

Thank you, I thought that it accepted only the numeric version.


So, I split this into the following PRs:

### Packaging tools - #2039
### ESLint and deps - #2033
### OLP SDKs - #2036 
### Tween - #2035 
### Testing tools - #2037 


I realized that the rest of the updates in this PR are mostly patches, so nothing that's worth dedicating a PR.
My suggestion is to close this one and move our attention to the ones above.
